library/L2_HAL/sensors/battery/ltc4150.hpp

```diff
@@ -4,95 +4,86 @@
+const float kCoulombsPerAh = 3600.0f;
 
 /// LTC4150 control driver for the LPC40xx and LPC17xx microcontrollers. It
 /// keeps track of a connected battery's current milliamp hours.
 class Ltc4150 : public CoulombCounter
 {
  public:
-  /// struct for holding the number of times an ISR invoked for a coulomb
-  /// counter
-  struct Battery_t
+  /// Defines what state the Polarity pin can be in.
+  enum class Polarity : uint8_t
   {
-    /// int variable holding the number of times an ISR invoked for a coulomb
-    /// counter
-    int num_ticks;
+    kDischarging,
+    kCharging,
   };
-
-  /// Class that handles the ticks from the LTC4150. The class uses a template
-  /// so there can be different handlers for different LTC4150s connected to a
-  /// board.
-  template <int id>
-  class TickHandler
-  {
-   public:
-    /// This member variable will hold the number of ticks sent from the
-    /// LTC4150. The interrupt handler increments this value by one every time
-    /// it is invoked.
-    inline static Battery_t battery_info;
-
-    /// handles the ISR by incrementing the num_ticks variable of TickHandler's
-    /// Battery_t
-    static void IsrHandler()
-    {
-      battery_info.num_ticks += 1;
-    }
-
-    /// @return the calculated milliamp hours based on the number of ticks.
-    /// 3600 is X 32.55 is kGvF and 0.00005 is kRsense
-    static float GetBatterymAh()
-    {
-      return static_cast<float>(static_cast<float>(battery_info.num_ticks) /
-                                (3600 * kGvf * kRsense));
-    }
-  };
-
-  /// @param isr - handler for coulomb counter ticks.
   /// @param int_pin - gpio pin for handling handling interrupts from the
-  /// LTC4150
+  ///        LTC4150
   /// @param pol - gpio pin to determine the polarity output of the LTC4150.
-  template <int id>
-  explicit constexpr Ltc4150(const TickHandler<id> & isr,
-                             Gpio & int_pin,
-                             Gpio & pol)
+  /// @param resistance - value of impedance represented in ohms for
+  ///        calculating battery charge
+  /// @param initial_polarity - value of the initial polarity when the LTC4150
+  /// is started up. The batery can either be charging or discharging.
+  explicit constexpr Ltc4150(sjsu::Gpio & int_pin,
+                             sjsu::Gpio & pol,
+                             units::impedance::ohm_t resistance,
+                             Polarity initial_polarity = Polarity::kDischarging)
       : int_pin_(int_pin),
-        pol_(pol),
-        get_mah_(isr.GetBatterymAh),
-        isr_handler_(isr.IsrHandler)
+        pol_pin_(pol),
+        resistance_(resistance),
+        pulses_(0),
+        polarity_(initial_polarity)
   {
   }
 
   /// Initialize hardware, setting pins as inputs and attaching ISR handlers to
   /// the interrupt pin.
-  void Initialize() const override
+  void Initialize() override
   {
-    pol_.SetAsInput();
+    pol_pin_.SetAsInput();
```

Comment:
One thing that should be done at initialization is that the pin should be read in order to figure out what the initial polarity is, rather than waiting for the polarity to change. In most applications, the polarity will be fixed from the point of power on.

Reply:
Done.